x-pack/plugins/cases/public/components/files/file_type.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import type {
+  ExternalReferenceAttachmentType,
+  ExternalReferenceAttachmentViewProps,
+} from '../../client/attachment_framework/types';
+
+import { FILE_ATTACHMENT_TYPE } from '../../../common/api';
+import { FileDownloadButtonIcon } from './file_download_button_icon';
+import { FileNameLink } from './file_name_link';
+import { FilePreview } from './file_preview';
+import * as i18n from './translations';
+import { isImage, isValidFileExternalReferenceMetadata } from './utils';
+
+interface FileAttachmentEventProps {
+  file: FileJSON;
+}
+
+const FileAttachmentEvent = ({ file }: FileAttachmentEventProps) => {
+  const [isPreviewVisible, setIsPreviewVisible] = useState(false);
+
+  const closePreview = () => setIsPreviewVisible(false);
+  const showPreview = () => setIsPreviewVisible(true);
```

Comment:
I think we use the same logic elsewhere. What about creating a hook (inside the files folder) called `useFilePreview` that encapsulates the previewing logic? For example:

```
const useFilePreview = () => {
  const [isPreviewVisible, setIsPreviewVisible] = useState(false);

  const closePreview = () => setIsPreviewVisible(false);
  const showPreview = () => setIsPreviewVisible(true);

  return { showPreview,  closePreview }
}

// consume it as
const { showPreview,  closePreview } = useFilePreview();

```